airflow/hooks/dbapi_hook.py

```diff
@@ -61,10 +61,17 @@ def __init__(self, *args, **kwargs):
             raise AirflowException("conn_name_attr is not defined")
         elif len(args) == 1:
             setattr(self, self.conn_name_attr, args[0])
-        elif self.conn_name_attr not in kwargs:
-            setattr(self, self.conn_name_attr, self.default_conn_name)
-        else:
+        elif self.conn_name_attr in kwargs:
             setattr(self, self.conn_name_attr, kwargs[self.conn_name_attr])
+        else:
+            setattr(self, self.conn_name_attr, self.default_conn_name)
```

Comment:
I change this un related cause it take me a little time to get the point, I think we should use `if condition ... else if ... else <default_value>` for more straightforward.

Reply:
```python
else:
    setattr(self, self.conn_name_attr, kwargs.get(self.conn_name_attr, self.default_conn_name)
```
instead of
```python
elif self.conn_name_attr in kwargs:
    setattr(self, self.conn_name_attr, kwargs[self.conn_name_attr])
else:
    setattr(self, self.conn_name_attr, self.default_conn_name)
```
WDYT?

Reply:
Oh, that much simple, but I have half-memories use `dict[key]` for exactly key as best practices. WDYT.

Reply:
Oh, that much simple, but I have half-memories use `dict[key]` for exactly key as best practices. WDYT.

Reply:
Add it